tests/unit/ml_handlers/test_pycaret.py

```diff
@@ -0,0 +1,214 @@
+import importlib
+import time
+from unittest.mock import patch
+import pandas as pd
+import pytest
+
+from mindsdb_sql import parse_sql
+from tests.unit.executor_test_base import BaseExecutorTest
+
+try:
+    importlib.import_module("pycaret")
+    PYCARET_INSTALLED = True
+except ImportError:
+    PYCARET_INSTALLED = False
+
+
+@pytest.mark.skipif(not PYCARET_INSTALLED, reason="pycaret is not installed")
+class TestPyCaret(BaseExecutorTest):
+
+    def wait_predictor(self, project, name):
+        done = False
+        for attempt in range(200):
+            ret = self.run_sql(
+                f"select * from {project}.models where name='{name}'"
+            )
+            if not ret.empty:
+                if ret['STATUS'][0] == 'complete':
+                    done = True
+                    break
+                elif ret['STATUS'][0] == 'error':
+                    break
+            time.sleep(0.5)
+        if not done:
+            raise RuntimeError("predictor wasn't created")
+
+    def run_sql(self, sql):
+        ret = self.command_executor.execute_command(
+            parse_sql(sql, dialect='mindsdb')
+        )
+        assert ret.error_code is None
+        if ret.data is not None:
+            columns = [
+                col.alias if col.alias is not None else col.name
+                for col in ret.columns
+            ]
+            return pd.DataFrame(ret.data, columns=columns)
+
+    @patch('mindsdb.integrations.handlers.postgres_handler.Handler')
+    def test_classifier(self, mock_handler):
+        iris_df = pd.read_csv('https://raw.githubusercontent.com/pycaret/pycaret/master/datasets/iris.csv')
```

Comment:
@aditya-azad - seems a bit unnecessary to download actual real data for unit tests. For speed and realiability reasons mostly.

I recon some fake/mock data should do the trick.

Reply:
Sure thing!